tiledb/ml/data_apis/pytorch_sparse.py

```diff
@@ -81,42 +75,46 @@ def __iter__(self):
             x_coords = []
             for i in range(0, self.x.schema.domain.ndim):
                 dim_name = self.x.schema.domain.dim(i).name
-                x_coords.append(np.array(x_batch[dim_name]))
+                x_coords.append(x_batch[dim_name])
 
-            # Normalise indices for torch.sparse.Tensor are batched with 0-index
-            x_coords[0] = np.vectorize(
-                lambda x: x - np.max(x_coords[0]) + self.batch_size - 1
-            )(x_coords[0])
+            # Normalise indices for torch.sparse.Tensor We want the coords indices in every iteration
+            # to be in the range of [0, self.batch_size] so the torch.sparse.Tensors can be created batch-wise.
+            # If we do not normalise the sparse tensor is being created but with a dimension [0, max(coord_index)],
+            # which is overkill
+            x_coords[0] -= x_coords[0].max() - self.batch_size + 1
```

Comment:
I'm still unsure about this line (and the respective one for y_coords)
- `We want the coords indices in every iteration to be in the range of [0, self.batch_size]`
   Please add an assertion for this after this line, I don't think it is satisfied as is now
- Why normalize only `x_coords[0]` and not all coords?

Reply:
I am writing also here for bookkeeping:

- In this iteration that we are working `x_coords[0].max() - x_coords[0].min() == batch_size` . So as you suggested I am converting it to the simpler representation `x_coords[0] -= x_coords[0].min()`
- The reason now why we normalise only `x_coords[0]` is because we are batching on "rows" dimension.